### ISSUE TITLE
Scripting feature for proxy maniOS app

### ISSUE THREAD

**Issue by whyike** [ProxymanApp/Proxyman#1449]
## Description

Any estimation when the scripting feature will arrive to the iOS application?

Thanks,
Mike.

**Comment by NghiaTranUIT** [comment]
It might take longer, since the Map Local and Breakpoint tools is iOS Road Map.

**Comment by NghiaTranUIT** [comment]
I suggest using the Scripting from the macOS app, since it's mature and well-developed 👍

**Comment by nihaals** [comment]
Now that Map Local and Breakpoints are released, is scripting the next large planned feature or is there something else with a higher priority? Would love to have scripting/Map Remote as it would essentially be the last piece for it to be at least possible to do almost anything from the iOS app.

**Comment by NghiaTranUIT** [comment]
@nihaals I highly suggest using the Scripting on macOS and debugs on your iPhone.

The problem is:
- iPhone has small screen, it's really difficult to write a JS Code -> I tried, but a touch screen is not designed for inputting a larger text
- Bad UX:

**Comment by NghiaTranUIT** [comment]
We consider porting the Map Remote and DNS Spoofing (Charles Proxy has this feature) to the iOS app soon.

**Comment by nihaals** [comment]
Regarding the UX of editing scripts, I'd get value out of the feature even if there was no editor whatsoever (I assume you wouldn't actually release this though). At the moment I do most debugging from the macOS app but after trying the iOS app more I've found it very convenient, particularly for using things already setup like Map Local. Once you've set up rules, all you need to do is start the VPN and do whatever you need to, instead of making sure you're with your Mac, start Proxyman, set the proxy settings up and then move on. I've mostly used the iOS app for Map Local which I don't have to change much and can just toggle on and off when I need to, and even with bad editing UX, scripts could fall into the same category.

The main place I've wished I had a script instead of map local is where I need to edit specific fields on a response and the response is mostly dynamic, and if the request is called a lot or you need to do a lot of replacements then breakpoints don't work either. I mentioned Map Remote since you could make your own proxy that essentially hosts all your scripts as a workaround, but obviously that's not ideal, and for what it's worth, I'd love to have scripts on iOS, even in some beta form that doesn't feel as polished as the Mac app.

**Comment by 1trackprojects1** [comment]
I agree with @nihaals as even if it doesn't have a fully-features editor and only an importing feature from the macOS editor it would be so helpful.

**Comment by NghiaTranUIT** [comment]
thanks for your input. Let's collect more request on this feature, so I can port the Scripting from macOS to iOS 👍

**Comment by gary-lo** [comment]
Any updates on this? Completely agree - simple import from macOS is all we need.

**Comment by NghiaTranUIT** [comment]
@gary-lo not yet, because I haven't seen many requests on this feature yet. I will keep this open, if more users prefer to have it, we will port the scripting tool to the iOS app 👍

**Comment by apclapp** [comment]
> @gary-lo not yet, because I haven't seen many requests on this feature yet. I will keep this open, if more users prefer to have it, we will port the scripting tool to the iOS app 👍

It doesn't seem very fair to advertise the feature as "coming soon" in the iOS app if it hasn't been decided whether the feature will be included or not. I purchased the iOS app seeing that 'coming soon' with the expectation that it was on some roadmap, even if it might not arrive soon. 

Perhaps indicate to the iOS users that they need to express interest if they would ever like the feature? I only came across this thread when looking for a timeline.

**Comment by gary-lo** [comment]
> We consider porting the Map Remote and DNS Spoofing (Charles Proxy has this feature) to the iOS app soon.

Getting map remote would also be great if that's easier and less work since map local is already there.

**Comment by adietrichs** [comment]
+1 for this feature request, would love to have scripting for iOS!

**Comment by lazur2006** [comment]
+1

**Comment by LazaroIII** [comment]
+1 This is the key missing feature.

**Comment by U5KUN** [comment]
+1 This is very important for people who don't have mac.

**Comment by samtvn** [comment]
+1

**Comment by arthurauffray** [comment]
+1 for scripting on ios! This would be amazing and crucial feature.
Thanks!

**Comment by NghiaTranUIT** [comment]
it seems that this feature is high demand 👍

**Comment by arthurauffray** [comment]
Amazing! Do you have an idea of a timeline? Also im wondering why you mark issues as done but dont close them?

Thanks!

**Comment by heyflorin** [comment]
Update on scripting for iOS/iPadOS? Looks like you have the demand for it. Thanks!

**Comment by NghiaTranUIT** [comment]
not yet due to an unsolved issue on iOS app (can't find a best way to support auto completion in the current Text Editor). It worked but bad experience.

May I ask @arthurauffray @heyflorin , what your use-case is? You would like to change the Request/Response headers ?

**Comment by arthurauffray** [comment]
Mainly response body, like rewriting/replacing/mtching specific data.

**Comment by thinhnq3009** [comment]
+1

**Comment by hibajbali** [comment]
+1

My use-case is slightly different. I'm not modifying headers, but rather triggering a side effect based on intercepted traffic if you want to keep into account

**Comment by NghiaTranUIT** [comment]
Hi folks, @michael-elkabetz @hibajbali @thinhnq3009 @arthurauffray @heyflorin @samtvn , .... There is a good news 👍 

I've finally introduced the Rewrite feature to Proxyman iOS
- Available on AppStore, v3.34.0 ✅ 
- It's easier to use the Scripting because small devices like iPhone is hard to write Javascript Code. I tried, it's bad
- Rewrite can help you do the following by GUI
1. Add/Remove/Replace Headers
2. Add/Remove/Replace Query
3. Change Status Code
4. Replace part of the Request/Response Body, or map with a new file
5. Change URL, Host, Path, ....

## Screenshots

<img width="1758" height="2346" alt="Image" src="https://github.com/user-attachments/assets/08ab06ca-09c1-4217-9b54-74b13a574dfc" />

https://github.com/user-attachments/assets/c88cfd71-fc0f-4c3f-88c4-6392245d7610

**Comment by NghiaTranUIT** [comment]
Appreciate your feedback 👍

**Comment by trinvh** [comment]
@NghiaTranUIT what's the best way to migrate an existing script on macOS app to iOS app?

**Comment by NghiaTranUIT** [comment]
@trinvh the iOS app doesn't have the Scripting Tool yet. You should use the Rewrite tool above, which is easier to use
